code/game/objects/items/books/skill_book.dm

```diff
@@ -633,11 +642,143 @@ MEDICAL
+	var/true_author //Used to keep track of who is actually writing the book.
+	var/list/failure_messages = list("Your hand slips and you accidentally rip your pen through several pages, ruining your hard work!","Your pen slips, dragging a haphazard line across both open pages! Now you need to do those again!")
+	var/list/progress_messages =list("Still quite a few blank pages left.","Feels like you're near halfway done.","You've made good progress.","Just needs a few finishing touches.","And then finish it. Done!") // Messages are in order of progress.
+	var/writing_time = 15 SECONDS // time it takes to write a segment of the book. This happens 6 times total
+
+//these all show up in the book fabricator
+/obj/item/book/skill/custom/black
+	icon_state = "book1"
+/obj/item/book/skill/custom/yellow
+	icon_state = "book3"
+/obj/item/book/skill/custom/blue
+	icon_state = "book4"
+/obj/item/book/skill/custom/green
+	icon_state = "book5"
+/obj/item/book/skill/custom/purple
+	icon_state = "book6"
+/obj/item/book/skill/custom/light_blue
+	icon_state = "book7"
+
+/obj/item/book/skill/custom/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/pen))
+
+		if(!user.skill_check(SKILL_LITERACY, SKILL_BASIC))
+			to_chat(user, SPAN_WARNING("You can't even read, yet you want to write a whole educational textbook?"))
+			return
+		if(!user.skill_check(SKILL_LITERACY, SKILL_PROF))
+			to_chat(user, SPAN_WARNING("You have no clue as to how to write an entire textbook in a way that is actually useful. Maybe a regular book would be better?"))
+			return
+
+		var/choice = input("What would you like to change?","Textbook editing") as null|anything in list("Title", "Author", skill_option_string)
+		switch(choice)
+
+			if("Title")
+				var/newtitle = reject_bad_text(sanitizeSafe(input("Write a new title:")))
+				if(!newtitle)
+					to_chat(usr, "The title is invalid.")
+					return
+				else
+					newtitle = usr.handle_writing_literacy(usr, newtitle)
+					if(newtitle)
+						title = newtitle
+						SetName(title)
+
+			if("Skill")
+				if(user.skillset.literacy_charges <= 0)
+					to_chat(user, SPAN_WARNING(pick("Your mind instantly recoils at the idea of having to write another entire textbook. No thank you!","You are far too mentally exhausted to write another textbook. Maybe another day.")))
+					return
+
+				//Choosing the skill
+				var/list/skill_choices = list()
+				for(var/decl/hierarchy/skill/S in GLOB.skills)
+					if(user.skill_check(S.type, SKILL_BASIC))
+						skill_choices += S
+				var/decl/hierarchy/skill/skill_choice = input(usr, "What subject does your textbook teach?", "Textbook skill selection") as null|anything in skill_choices
+				if(!skill_choice)
+					to_chat(user, SPAN_WARNING("Textbook skill selection cancelled."))
+					return
+				var/newskill = skill_choice.type
+
+				//Choosing the level
+				var/list/skill_levels = list("Basic" = SKILL_BASIC, "Trained" = SKILL_ADEPT, "Experienced" = SKILL_EXPERT, "Master" = SKILL_PROF)
```

Comment:
Oh. That's a good point.
Though, I'm not sure how to use the skill's levels list and have them correspond to values though, since the associated values are strings. I'm not actually very on top of using lists at all.

Reply:
You'd just need to use `skill_choice.levels` instead of declaring your own, as `skill_choice` has a list defined on it.

Reply:
Having the weirdest issue with this. I've done what I thought would be the best way to do this... and the skill datums are being deleted, and not just removed from what I thought I made as a temporary list. Viewing the skill panel shows the skill levels that get removed from selection lists as simply not being present.
```
				//Choosing the skill
				var/list/skill_choices = list()
				for(var/decl/hierarchy/skill/S in GLOB.skills)
					if(user.skill_check(S.type, SKILL_BASIC))
						LAZYADD(skill_choices, S)
				var/decl/hierarchy/skill/skill_choice = input(user, "What subject does your textbook teach?", "Textbook skill selection") as null|anything in skill_choices
				if(!can_write(user,W) || progress == SKILLBOOK_PROG_FINISH)
					return
				if(!skill_choice)
					to_chat(user, SPAN_WARNING("Textbook skill selection cancelled."))
					return
				var/newskill = skill_choice.type

				//Choosing the level
				var/list/skill_levels = skill_choice.levels
				for(var/SL in skill_levels)
					if(!user.skill_check(newskill, skill_levels.Find(SL)))
						LAZYREMOVE(skill_levels, SL)
					else
						skill_levels[SL] = skill_levels.Find(SL) + 1
				LAZYREMOVE(skill_levels,skill_levels[1])
				var/newskill_level = input(user, "What level of education does it provide?","Textbook skill level") as null|anything in skill_levels
```

Reply:
Woops nevermind. Good now.